src/ucp/core/ucp_rkey.c

```diff
@@ -888,9 +888,9 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_ep_rkey_unpack_internal,
         if (status == UCS_OK) {
             ucs_trace("rkey[%d] for remote md %d is 0x%lx", rkey_index,
                       remote_md_index, tl_rkey->rkey.rkey);
+            rkey->md_map |= UCS_BIT(remote_md_index);
```

Comment:
what if we revert rkey->md_map in case of error? to avoid extra bit op for each md. rkey_unpack happends during datapath such as rndv protocol.